packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -201,17 +217,22 @@ export function makeRumPublicApi(
     }),
 
     setUser: monitor((newUser: User) => {
-      const sanitizedUser = sanitizeUser(newUser)
-      if (sanitizedUser) {
-        user = sanitizedUser
-      } else {
+      if (typeof newUser !== 'object' || !newUser) {
         display.error('Unsupported user:', newUser)
+      } else {
+        userContextManager.setContext(sanitizeUser(newUser as Context))
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I's a bit a shame to perform two deep clones here. What about doing a shallow clone just for the sanitization (in order to not overwriting customer `id`, `name` or `email`) and let the `setContext` perform the deep clone?